### ISSUE TITLE
Error: Failed to find a python version from target process

### ISSUE THREAD

**Issue by ahmed-shariff** [benfred/py-spy#397]
For pretty much any command i run (top, record and dump) i get the following:

```
Error: Failed to find a python version from target process
```

I've installed py-spy through poetry (pip). I am trying running it through both the `poetry run py-spy ..` as well as activate the env and run `py-spy ...`

OS: Windows 10
Python: 3.9.4

**Comment by Jongy** [comment]
Which py-spy version is it? (displayed when you run plain `py-spy`, first line).

Also - can you please run with `RUST_LOG=debug` (for example - `sudo RUST_LOG=debug py-spy ...`) and post the outputs?

**Comment by benfred** [comment]
Can you also try profiling with subprocesses (w/ the ```--subprocesses``` flag) ? We've seen issues like https://github.com/benfred/py-spy/issues/81 and https://github.com/benfred/py-spy/issues/216 on windows with virtual environments before - where the python process is created as a subprocess

**Comment by ahmed-shariff** [comment]
Thanks for the quick reply guys.

The version I have is `0.3.7`.

and oddly, after reinstalling the package, the commands seems to work. I am not sure what changed though. I'll open the issue again if I run into any problems.

**Comment by rhughesjr** [comment]
FYI I just ran into this error and fortunately found this solution. I'm running Win10 and a venv.

Maybe this would be good to stick in the faq's?

**Comment by gobater** [comment]
This problem is not yet fixed in py-spy 0.3.14 on windows when profiling native code.
Combination of "--native and -s" options throws the following message:
```
Can't get native stack traces with the ---subprocesses option on windows.
```

**Comment by AdityaSoni19031997** [comment]
When I ran with DEBUG flag, I got this at the very end -: 

```
[2023-11-20T04:32:28.245220381Z INFO  py_spy::python_spy] Getting version from python binary BSS
[2023-11-20T04:32:28.245279172Z INFO  py_spy::python_spy] Failed to get version from BSS section: failed to find version string
```

**Comment by AcidInvader** [comment]
I'v got the issue: `Error: Failed to open process - check if it is running.
`
 when i run this command `py-spy record -o profile.svg --pid 8039`.  Please help me how can i solve it?

**Comment by cemlyn007** [comment]
In my case, it looks it struggles to get the extract the version from the path:
```
[2024-06-17T11:12:05.444696000Z INFO  py_spy::python_spy] got symbol Py_GetVersion.version (0x00000001037ae650) from libpython binary
[2024-06-17T11:12:05.444702000Z INFO  py_spy::python_spy] Getting version from symbol address
[2024-06-17T11:12:05.444717000Z INFO  py_spy::python_spy] Getting version from python binary BSS
[2024-06-17T11:12:05.444720000Z INFO  py_spy::python_spy] Failed to get version from BSS section: failed to find version string
[2024-06-17T11:12:05.444722000Z INFO  py_spy::python_spy] Getting version from libpython BSS
[2024-06-17T11:12:05.444732000Z INFO  py_spy::python_spy] Failed to get version from libpython BSS section: failed to find version string
[2024-06-17T11:12:05.444733000Z INFO  py_spy::python_spy] Trying to get version from path: /opt/homebrew/Cellar/python@3.10/3.10.14/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python
```

**Comment by mrjazz** [comment]
Have the same problem, py-spy version is 0.3.14, MacOS version is Sonnoma 14.5

**Comment by aditya-sengupta** [comment]
Same problem, py-spy 0.3.14, MacOS Ventura 13.2.1

**Comment by amenasria** [comment]
For macOS folks, just spent 3 hours debugging this 🫡 Here's how I fixed it:

TL; DR: I uninstalled the Python I installed through `brew` or the Python installer (because they might be conflict with reading memory from those) and used [pyenv](https://github.com/pyenv/pyenv) to install a local python (see script below). After that it worked !
```bash
# Uninstall other Python binaries through brew or rm
brew uninstall python@3.11
# Setup pyenv
brew update
brew install pyenv
echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.zshrc
echo '[[ -d $PYENV_ROOT/bin ]] && export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.zshrc
echo 'eval "$(pyenv init - zsh)"' >> ~/.zshrc
#  Restart your shell
exec "$SHELL"
# Install Python
pyenv install 3.11.8
# Use it globally
pyenv global 3.11.8
```

Longer explanation ([from this section of py-spy README](https://github.com/benfred/py-spy/blob/1fa3a6ded252d7c1c0ff974a4fcd1af67a1577cf/README.md#why-am-i-having-issues-profiling-usrbinpython-on-osx)): macOS [SIP](https://en.wikipedia.org/wiki/System_Integrity_Protection) is preventing all users from reading memory from any binary located in protected directories (such as `/usr/bin` maybe containing your Python binary). So you should either:
- disable SIP (kinda risky),
- install a Python that won't be protected by SIP(recommended).

Hope this helped !

**Comment by ujaandas** [comment]
I'm trying to use py-spy on a locally compiled Cpython but get the same error. Why is this closed? I've even tried changing define fields in patchlevel.h, but to no avail.

**Comment by alanjds** [comment]
I am getting the same issue on Linux with Python 3.13.11 inside a virtualenv:

The error is `failed to get gil_thread_id`. May this is because 3.13t have no GIL enabled?

```
$ RUST_LOG=INFO py-spy record -o perf/spy-asyncio-proto.speedscope.json --format speedscope --native --subprocesses -- python /var/home/alanjds/src/tokioloop/benchmarks/server.py --loop asyncio --proto
[2026-02-11T13:45:06.191171294Z INFO  py_spy::config] Command line args: ArgMatches { args: {}, subcommand: Some(SubCommand { id: [hash: FA6893386E553029], name: "record", matches: ArgMatches { args: {[hash: 94949CDCC1D03F21]: MatchedArg { occurs: 1, source: Some(CommandLine), indices: [2], type_id: Some(TypeId(0xb4ba764d8c901af36906977ef03c5b82)), vals: [[AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }]], raw_vals: [["perf/spy-asyncio-proto.speedscope.json"]], ignore_case: false }, [hash: 66D7DFD73E3FE4B7]: MatchedArg { occurs: 1, source: Some(CommandLine), indices: [4], type_id: Some(TypeId(0xb4ba764d8c901af36906977ef03c5b82)), vals: [[AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }]], raw_vals: [["speedscope"]], ignore_case: true }, [hash: 3F6365685C41EC2F]: MatchedArg { occurs: 1, source: Some(CommandLine), indices: [5], type_id: Some(TypeId(0xb4ba764d8c901af36906977ef03c5b82)), vals: [[]], raw_vals: [[]], ignore_case: false }, [hash: 721FFA7420DEDCF]: MatchedArg { occurs: 1, source: Some(CommandLine), indices: [6], type_id: Some(TypeId(0xb4ba764d8c901af36906977ef03c5b82)), vals: [[]], raw_vals: [[]], ignore_case: false }, [hash: DFA48303AEDA2B21]: MatchedArg { occurs: 5, source: Some(CommandLine), indices: [7, 8, 9, 10, 11], type_id: Some(TypeId(0xb4ba764d8c901af36906977ef03c5b82)), vals: [[AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }, AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }, AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }, AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }, AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }]], raw_vals: [["python", "/var/home/alanjds/src/tokioloop/benchmarks/server.py", "--loop", "asyncio", "--proto"]], ignore_case: false }, [hash: BD514D8BE9BFDED6]: MatchedArg { occurs: 0, source: Some(DefaultValue), indices: [12], type_id: Some(TypeId(0xb4ba764d8c901af36906977ef03c5b82)), vals: [[AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }]], raw_vals: [["unlimited"]], ignore_case: false }, [hash: AAFAF59EB7E2AC8]: MatchedArg { occurs: 0, source: Some(DefaultValue), indices: [13], type_id: Some(TypeId(0xb4ba764d8c901af36906977ef03c5b82)), vals: [[AnyValue { inner: TypeId(0xb4ba764d8c901af36906977ef03c5b82) }]], raw_vals: [["100"]], ignore_case: false }}, subcommand: None } }) }
[2026-02-11T13:45:06.192320647Z INFO  py_spy::python_process_info] Got virtual memory maps from pid 1228093:
[2026-02-11T13:45:06.208568120Z INFO  py_spy::python_process_info] Getting version from python binary BSS
[2026-02-11T13:45:06.210662833Z INFO  py_spy::version] Found matching version string '3.13.11 experimental free-threading build (main, Dec 17 2025, 21:08:58) '
[2026-02-11T13:45:06.210689272Z INFO  py_spy::python_spy] python version 3.13.11 detected
[2026-02-11T13:45:06.210694492Z INFO  py_spy::python_process_info] got symbol _PyRuntime (0x00000000014a1a80) from python binary
[2026-02-11T13:45:06.210780019Z INFO  py_spy::python_spy] Found interpreter at 0x00000000014bd2c0
[2026-02-11T13:45:06.210819838Z INFO  remoteprocess::linux::symbolication] reloading process binaries
[2026-02-11T13:45:06.211219825Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.214665866Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.217382149Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libc.so.6
[2026-02-11T13:45:06.217603422Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libm.so.6
[2026-02-11T13:45:06.217693519Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/librt.so.1
[2026-02-11T13:45:06.217724358Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libutil.so.1
[2026-02-11T13:45:06.217748048Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libdl.so.2
[2026-02-11T13:45:06.217772897Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libpthread.so.0
[2026-02-11T13:45:06.217796056Z INFO  remoteprocess::linux::symbolication] loading debug info from [vdso]
[2026-02-11T13:45:06.217830025Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
[2026-02-11T13:45:06.222252784Z INFO  py_spy::python_spy] Failed to connect to process, retrying. Error: failed to get gil_thread_id
[2026-02-11T13:45:06.242585507Z INFO  py_spy::python_process_info] Got virtual memory maps from pid 1228093:
[2026-02-11T13:45:06.256152156Z INFO  py_spy::python_process_info] Getting version from python binary BSS
[2026-02-11T13:45:06.256384998Z INFO  py_spy::version] Found matching version string '3.13.11 experimental free-threading build (main, Dec 17 2025, 21:08:58) '
[2026-02-11T13:45:06.256407637Z INFO  py_spy::python_spy] python version 3.13.11 detected
[2026-02-11T13:45:06.256413557Z INFO  py_spy::python_process_info] got symbol _PyRuntime (0x00000000014a1a80) from python binary
[2026-02-11T13:45:06.256438536Z WARN  py_spy::python_process_info] Interpreter address from _PyRuntime symbol is invalid 00000000014bd2c0
[2026-02-11T13:45:06.256443636Z INFO  py_spy::python_process_info] Failed to find runtime address from symbols, scanning BSS section from main binary
[2026-02-11T13:45:06.256588212Z INFO  py_spy::python_spy] Found interpreter at 0x00000000014bd2c0
[2026-02-11T13:45:06.256607601Z INFO  remoteprocess::linux::symbolication] reloading process binaries
using asyncio loop
[2026-02-11T13:45:06.256715508Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
serving on: ('127.0.0.1', 25000)
using simple protocol
PID 1228093
[2026-02-11T13:45:06.259614285Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.262298290Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libc.so.6
[2026-02-11T13:45:06.262514393Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libm.so.6
[2026-02-11T13:45:06.262617020Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/librt.so.1
[2026-02-11T13:45:06.262645519Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libutil.so.1
[2026-02-11T13:45:06.262670328Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libdl.so.2
[2026-02-11T13:45:06.262693977Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libpthread.so.0
[2026-02-11T13:45:06.262718447Z INFO  remoteprocess::linux::symbolication] loading debug info from [vdso]
[2026-02-11T13:45:06.262743896Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
[2026-02-11T13:45:06.264792911Z INFO  py_spy::python_spy] Failed to connect to process, retrying. Error: failed to get gil_thread_id
[2026-02-11T13:45:06.285063295Z INFO  py_spy::python_process_info] Got virtual memory maps from pid 1228093:
[2026-02-11T13:45:06.298469759Z INFO  py_spy::python_process_info] Getting version from python binary BSS
[2026-02-11T13:45:06.298658323Z INFO  py_spy::version] Found matching version string '3.13.11 experimental free-threading build (main, Dec 17 2025, 21:08:58) '
[2026-02-11T13:45:06.298677852Z INFO  py_spy::python_spy] python version 3.13.11 detected
[2026-02-11T13:45:06.298682202Z INFO  py_spy::python_process_info] got symbol _PyRuntime (0x00000000014a1a80) from python binary
[2026-02-11T13:45:06.298712111Z INFO  py_spy::python_spy] Found interpreter at 0x00000000014bd2c0
[2026-02-11T13:45:06.298724241Z INFO  remoteprocess::linux::symbolication] reloading process binaries
[2026-02-11T13:45:06.298821748Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.301583710Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.304141308Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libc.so.6
[2026-02-11T13:45:06.304350282Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libm.so.6
[2026-02-11T13:45:06.304443319Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/librt.so.1
[2026-02-11T13:45:06.304466998Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libutil.so.1
[2026-02-11T13:45:06.304492607Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libdl.so.2
[2026-02-11T13:45:06.304514916Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libpthread.so.0
[2026-02-11T13:45:06.304535686Z INFO  remoteprocess::linux::symbolication] loading debug info from [vdso]
[2026-02-11T13:45:06.304556895Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
[2026-02-11T13:45:06.306536862Z INFO  py_spy::python_spy] Failed to connect to process, retrying. Error: failed to get gil_thread_id
[2026-02-11T13:45:06.326818847Z INFO  py_spy::python_process_info] Got virtual memory maps from pid 1228093:
[2026-02-11T13:45:06.340535930Z INFO  py_spy::python_process_info] Getting version from python binary BSS
[2026-02-11T13:45:06.340716444Z INFO  py_spy::version] Found matching version string '3.13.11 experimental free-threading build (main, Dec 17 2025, 21:08:58) '
[2026-02-11T13:45:06.340741384Z INFO  py_spy::python_spy] python version 3.13.11 detected
[2026-02-11T13:45:06.340746243Z INFO  py_spy::python_process_info] got symbol _PyRuntime (0x00000000014a1a80) from python binary
[2026-02-11T13:45:06.340775293Z INFO  py_spy::python_spy] Found interpreter at 0x00000000014bd2c0
[2026-02-11T13:45:06.340789252Z INFO  remoteprocess::linux::symbolication] reloading process binaries
[2026-02-11T13:45:06.340883879Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.343645751Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.346192250Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libc.so.6
[2026-02-11T13:45:06.346390464Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libm.so.6
[2026-02-11T13:45:06.346488261Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/librt.so.1
[2026-02-11T13:45:06.346514190Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libutil.so.1
[2026-02-11T13:45:06.346535719Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libdl.so.2
[2026-02-11T13:45:06.346557589Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libpthread.so.0
[2026-02-11T13:45:06.346578128Z INFO  remoteprocess::linux::symbolication] loading debug info from [vdso]
[2026-02-11T13:45:06.346602687Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
[2026-02-11T13:45:06.348527996Z INFO  py_spy::python_spy] Failed to connect to process, retrying. Error: failed to get gil_thread_id
[2026-02-11T13:45:06.368684124Z INFO  py_spy::python_process_info] Got virtual memory maps from pid 1228093:
[2026-02-11T13:45:06.382081348Z INFO  py_spy::python_process_info] Getting version from python binary BSS
[2026-02-11T13:45:06.382264022Z INFO  py_spy::version] Found matching version string '3.13.11 experimental free-threading build (main, Dec 17 2025, 21:08:58) '
[2026-02-11T13:45:06.382284692Z INFO  py_spy::python_spy] python version 3.13.11 detected
[2026-02-11T13:45:06.382289251Z INFO  py_spy::python_process_info] got symbol _PyRuntime (0x00000000014a1a80) from python binary
[2026-02-11T13:45:06.382316261Z INFO  py_spy::python_spy] Found interpreter at 0x00000000014bd2c0
[2026-02-11T13:45:06.382328930Z INFO  remoteprocess::linux::symbolication] reloading process binaries
[2026-02-11T13:45:06.382431457Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.385244817Z INFO  remoteprocess::linux::symbolication] loading debug info from /home/alanjds/.local/share/uv/python/cpython-3.13.11+freethreaded-linux-x86_64-gnu/bin/python3.13t
[2026-02-11T13:45:06.387837535Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libc.so.6
[2026-02-11T13:45:06.388041288Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libm.so.6
[2026-02-11T13:45:06.388132025Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/librt.so.1
[2026-02-11T13:45:06.388156155Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libutil.so.1
[2026-02-11T13:45:06.388180684Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libdl.so.2
[2026-02-11T13:45:06.388199653Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/libpthread.so.0
[2026-02-11T13:45:06.388220413Z INFO  remoteprocess::linux::symbolication] loading debug info from [vdso]
[2026-02-11T13:45:06.388241762Z INFO  remoteprocess::linux::symbolication] loading debug info from /usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
[2026-02-11T13:45:06.390159681Z WARN  py_spy::sampler] Failed to profile python from process 1228093: failed to get gil_thread_id
[2026-02-11T13:45:06.498080407Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
[2026-02-11T13:45:06.605626574Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
[2026-02-11T13:45:06.713274779Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
[2026-02-11T13:45:06.820698390Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
[2026-02-11T13:45:06.929047242Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
[2026-02-11T13:45:07.036415315Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
[2026-02-11T13:45:07.143959823Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
[2026-02-11T13:45:07.251236279Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
[2026-02-11T13:45:07.358894463Z WARN  py_spy::sampler] Failed to get initialization status from PythonSpyThread: receiving on a closed channel
Error: No python processes found in process 1228093 or any of its subprocesses```

**Comment by alanjds** [comment]
Looks like Python 3.13.11 free-threading is not supported right now. Can this be confirmed?

**Comment by Dadaskis** [comment]
```
PS D:\PythonProjects\ozon_parser> py-spy top --pid 9044
Error: Failed to find python version from target process
PS D:\PythonProjects\ozon_parser> py-spy top --pid 9044 --subprocesses
Error: No python processes found in process 9044 or any of its subprocesses
PS D:\PythonProjects\ozon_parser> py-spy top -- python .\main.py      

Error: Failed to find python version from target process
PS D:\PythonProjects\ozon_parser> py-spy top -s python .\main.py
18-03-2026 19:24:02 - Main - INFO - Ozon Scraper start!
18-03-2026 19:24:02 - OzonScraper - INFO - Start - Beginning.
18-03-2026 19:24:02 - OzonSearcher - INFO - Start - Beginning.
18-03-2026 19:24:02 - OzonSearcher - INFO - Starting Playwright...
18-03-2026 19:24:02 - OzonDescriptionPeeker - INFO - Start - Beginning.
18-03-2026 19:24:02 - OzonDescriptionPeeker - INFO - Starting Playwright...
18-03-2026 19:24:02 - OzonDescriptionPeeker - INFO - Playwright started!
18-03-2026 19:24:02 - OzonDescriptionPeeker - INFO - Launching a browser...
18-03-2026 19:24:02 - OzonSearcher - INFO - Playwright started!
18-03-2026 19:24:02 - OzonSearcher - INFO - Launching a browser...
18-03-2026 19:24:03 - OzonDescriptionPeeker - INFO - Browser is intact!
18-03-2026 19:24:03 - OzonDescriptionPeeker - INFO - Creating a page...
18-03-2026 19:24:03 - OzonSearcher - INFO - Browser is intact!
18-03-2026 19:24:03 - OzonSearcher - INFO - Creating a page...
18-03-2026 19:24:03 - OzonSearcher - INFO - Going to ozon.by...
18-03-2026 19:24:03 - OzonDescriptionPeeker - INFO - Going to ozon.by...
18-03-2026 19:24:04 - OzonSearcher - INFO - Waiting for the page to pass the bot test - 3 seconds...
18-03-2026 19:24:04 - OzonDescriptionPeeker - INFO - Waiting for the page to pass the bot test - 3 seconds...

Error: No python processes found in process 7360 or any of its subprocesses
PS D:\PythonProjects\ozon_parser> 
```

2026, reporting that this issue needs to be opened :)

Python 3.14.0
Windows 10
py-spy 0.4.1
And yeah the target script had an async event loop running with like 2 coroutines working at the same time, not sure if it helps - just letting you know.

---

Update: Doesn't work on Arch Linux either

```
[dadaskis@Dadaskis ozon_parser]$ sudo py-spy top --pid 13110
Error: Failed to find python version from target process
[dadaskis@Dadaskis ozon_parser]$ 
```

FastFetch:
```
                  -`                     dadaskis@Dadaskis
                 .o+`                    -----------------
                `ooo/                    OS: Arch Linux x86_64
               `+oooo:                   Kernel: Linux 6.19.8-arch1-1
              `+oooooo:                  Uptime: 1 hour, 11 mins
              -+oooooo+:                 Packages: 970 (pacman)
            `/:-:++oooo+:                Shell: bash 5.3.9
           `/++++/+++++++:               Display (AE24AVB): 1920x1080 in 24", 75 Hz [External]
          `/++++++++++++++:              DE: KDE Plasma 6.6.3
         `/+++ooooooooooooo/`            WM: KWin (Wayland)
        ./ooosssso++osssssso+`           WM Theme: Breeze
       .oossssso-````/ossssss+`          Theme: Breeze (Dark) [Qt], Breeze-Dark [GTK2], Breeze [GTK3]
      -osssssso.      :ssssssso.         Icons: breeze-dark [Qt], breeze-dark [GTK2/3/4]
     :osssssss/        osssso+++.        Font: Noto Sans (10pt) [Qt], Noto Sans (10pt) [GTK2/3/4]
    /ossssssss/        +ssssooo/-        Cursor: breeze (24px)
  `/ossssso+/:-        -:/+osssso+-      Terminal: konsole 25.12.3
 `+sso+:-`                 `.-/+oso:     CPU: Intel(R) Core(TM) i5-7400 (4) @ 3.50 GHz
`++:.                           `-/+/    GPU: AMD Radeon RX 580 2048SP [Discrete]
.`                                 `/    Memory: 7.66 GiB / 15.58 GiB (49%)
                                         Swap: 0 B / 7.79 GiB (0%)
                                         Disk (/): 26.39 GiB / 456.39 GiB (6%) - ext4
                                         Disk (/mnt/windows_drive): 1.48 TiB / 1.82 TiB (81%) - fuseblk
                                         Local IP (enp0s31f6): 192.168.0.100/24
                                         Locale: en_US.UTF-8

```
